rust/src/cli.rs

```diff
@@ -102,43 +124,44 @@ pub fn parse_args() -> Args {
 pub async fn store(
     vault: &Vault,
     key: Option<String>,
-    value: Option<String>,
+    value_positional: Option<String>,
     file: Option<String>,
-    value_opt: Option<String>,
+    value_argument: Option<String>,
     overwrite: bool,
 ) -> Result<()> {
     let key = {
-        if let Some(key) = &key {
+        if let Some(key) = key {
             key
         } else if let Some(file_name) = &file {
-            match file_name.as_str() {
-                "-" => anyhow::bail!("Key cannot be empty when reading from stdin".red()),
-                _ => file_name,
+            if file_name == "-" {
+                anyhow::bail!("Key cannot be empty when reading from stdin".red())
             }
+            let key = get_filename_from_path(file_name)?;
+            println!("Using filename as key: '{key}'");
+            key
         } else {
             anyhow::bail!(
                 "Empty key and no {} flag provided, provide at least one of these",
-                "-f".yellow().bold()
+                "--file".yellow().bold()
             )
         }
     };
 
     let data = {
-        if let Some(value) = value.or(value_opt) {
-            value.clone()
+        if let Some(value) = value_positional.or(value_argument) {
```

Comment:
I would personally swap these two around, meaning explicit `--value` will override a positional argument, since it is more probable that a positional arg was a mistake when specifying `-v` / `--value`